pkg/kubelet/prober/worker_test.go

```diff
@@ -508,6 +502,6 @@ func TestStartupProbeDisabledByStarted(t *testing.T) {
 	// startupProbe fails, but is disabled
 	m.prober.exec = fakeExecProber{probe.Failure, nil}
 	msg = "Started, probe failure, result success"
-	expectStop(t, w, w.doProbe(), msg)
+	expectContinue(t, w, w.doProbe(), msg)
```

Comment:
if this was so easy to miss, does it deserve a real e2e test for actual restarting container?

Reply:
Probably... I think the main issue was that I mixed up pod lifecycle and container lifecycle while coding, and then tried too hard to convince you and Elena :-(
The worst is that we went GA in 1.20 and the cherry-pick broke the probe in the latest 1.20 patch release...

Reply:
Let's add an e2e in a followup.